tracing-appender/src/rolling/builder.rs

```diff
@@ -0,0 +1,168 @@
+            .to_string();
+        Builder {
+            log_directory,
+            log_filename_prefix,
+            rotation: None,
+            #[cfg(feature = "compression_gzip")]
+            compression: None,
+        }
+    }
+
+    /// Configures when log files will be rotated.
+    ///
+    /// By default, no rotation will occur.
+    pub fn rotation(mut self, rotation: Rotation) -> Self {
+        self.rotation = Some(rotation);
+        self
+    }
+
+    /// Sets compression level
+    #[cfg(feature = "compression_gzip")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "compression_gzip")))]
+    pub fn compression(mut self, compression: CompressionOption) -> Self {
+        self.compression = Some(compression.into());
+        self
+    }
+
+    pub(crate) fn get_extension(&self) -> Option<String> {
+        #[cfg(feature = "compression_gzip")]
+        if let Some(compression) = self.compression.clone() {
+            compression.extension().map(|v| v.to_string())
+        } else {
+            None
+        }
+
+        #[cfg(not(feature = "compression_gzip"))]
+        None
+    }
+
+    /// Returns a new [`RollingFileAppender`] with the configuration defined by this builder.
+    pub fn build(self) -> RollingFileAppender {
+        let now = OffsetDateTime::now_utc();
+        let rotation = self.rotation.clone().unwrap_or(Rotation::NEVER);
+        let extension = self.get_extension();
+        let filename = rotation.join_date(self.log_filename_prefix.as_str(), &now, extension);
+        let next_date = rotation.next_date(&now);
+
+        #[cfg(not(feature = "compression_gzip"))]
+        let writer = self.create_file_writer(filename.as_str());
+
+        #[cfg(feature = "compression_gzip")]
+        let writer = if let Some(compression) = self.compression.clone() {
+            RwLock::new(
+                WriterChannel::new_with_compression(
+                    self.log_directory.as_str(),
+                    &filename,
+                    compression,
+                )
+                .unwrap(),
+            )
+        } else {
+            self.create_file_writer(filename.as_str())
+        };
+
+        let next_date = AtomicUsize::new(
+            next_date
+                .map(|date| date.unix_timestamp() as usize)
+                .unwrap_or(0),
+        );
+
+        RollingFileAppender {
+            state: Inner {
+                log_directory: self.log_directory,
+                log_filename_prefix: self.log_filename_prefix,
+                next_date,
+                rotation,
+                #[cfg(feature = "compression_gzip")]
+                compression: self.compression,
+            },
+            writer,
+        }
+    }
+
+    fn create_file_writer(&self, filename: &str) -> RwLock<WriterChannel> {
+        let a = RwLock::new(WriterChannel::File(
+            create_writer_file(self.log_directory.as_str(), filename)
+                .expect("failed to create appender"),
+        ));
+        a
```

Comment:
you don't really need the `a` variable here since we are just going to return it right after

Reply:
Fixed will push soon.